modules/programs/senpai.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.senpai;
+  cfgType = pkgs.formats.yaml { };
+in {
+  options.programs.senpai = {
+    enable = mkEnableOption "senpai";
+    config = mkOption {
+      type = types.submodule {
+        freeformType = cfgType.type;
+        options = {
+          addr = mkOption {
+            type = types.str;
+            description = ''
+              The address (host[:port]) of the IRC server. senpai uses TLS connections
+              by default unless you specify no-tls option. TLS connections default
+              to port 6697, plain-text use port 6667.
+            '';
+          };
+          nick = mkOption {
+            type = types.str;
+            description = ''
+              Your nickname, sent with a NICK IRC message. It mustn't contain spaces
+              or colons (:).
+            '';
+          };
+        };
+      };
+      description = ''
+        Configuration for senpai. For a list of options, see
+        <citerefentry><refentrytitle>senpai</refentrytitle>
+        <manvolnum>5</manvolnum></citerefentry>.
+      '';
+    };
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Lines added verbatim to the configuration file.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.senpai ];
+    xdg.configFile."senpai/senpai.yaml".text = ''
+      ${builtins.toJSON cfg.config}
+      ${cfg.extraConfig}
```

Comment:
one nice thing about getting rid of this is that you can just use: `cfgFmt.generate "senpai.yaml" cfg.config` directly.